### PR TITLE
Add Task Use Case Tests

### LINKED ISSUE #65: [Feature 8.1] Add Task Tests

### PR DESCRIPTION
**Pull request by bmguiler**
I have written passing tests for the Add Task Use Case.

When writing tests, I found a bug in the AddTaskUseCase class, so that bug fix is included in this PR.

The naming convention for my test class is set to match the others, however the test packaging still needs to be refactored.

### CODE REVIEW

**Comment by bmguiler** [comment]
jk some of the tests are failing on gradle so i need to figure that out first

File: src/main/java/use_cases/add_task/AddTaskUseCase.java [hunk 1]
```diff
@@ -48,8 +48,9 @@ public AddTaskUseCase(AddTaskOutputBoundary presenter,
     public void addTask(String taskName, String taskDescription, int parentId,
                         int curriculumId) {
 
-        Task newTask = this.taskFactory.create(taskName, taskDescription);
-        TaskTree newTaskTree = this.taskTreeFactory.create();
+        TaskTree newTaskTree = this.taskTreeFactory.createWithTask(taskName, taskDescription);
+        Task newTask = newTaskTree.getTask();
```
Comment (Bhan-Aayush) [inline src/main/java/use_cases/add_task/AddTaskUseCase.java:52]: You have the correct name of (taskName, taskDescription), where you get to populate the tasktree. Good use of clean architecture as well, once I get Oswin's Ui your addTask feature should also work with my tree.

**Review by Bhan-Aayush — DISMISSED** [review]

**Comment by Bhan-Aayush** [comment]
Overall, I see you are fixing your tests, but I believe you are on the right track, keep up the good work. Once I am done implementing other parts of the code in my TaskTreeUI(by including your addtasktree method), it will populate the tasktree and your tests will run smoothly.

**Review by Eggathin — DISMISSED** [review]
Contains bugs that are expected to be fixed with a future PR. Otherwise strong test coverage

**Comment by bmguiler** [comment]
The tests pass now; the issue was that having the curriculum and schedule as class variables, each test compounded the effects of the last one. so now i have schedule and curriculum variables for each test method and it works.

**Review by Eggathin — APPROVED** [review]
Re-approving the pull request now that the tests pass

**Comment by Bhan-Aayush** [comment]
Everything looks fine now. Good Job!

**Review by Bhan-Aayush — APPROVED** [review]

**Review by dadler83 — APPROVED** [review]
Nice bug catch, good to catch these before we merge. Testing looks thorough! Approved

**Pull request merged**